server/game/cards/01_SOR/leaders/BobaFettCollectingTheBounty.ts

```diff
@@ -0,0 +1,35 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { LeaderUnitCard } from '../../../core/card/LeaderUnitCard';
+
+export default class BobaFettCollectingTheBounty extends LeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '4626028465',
+            internalName: 'boba-fett#collecting-the-bounty',
+        };
+    }
+
+    protected override setupLeaderSideAbilities() {
+        this.addTriggeredAbility({
+            title: 'Ready a resource',
+            cost: [AbilityHelper.costs.exhaustSelf()],
+            when: {
+                onCardReturnedToHand: (event, context) => event.card.isUnit() && event.card.controller !== context.source.controller && context.source.controller.resources.some((resource) => resource.exhausted),
+                onCardDefeated: (event, context) => event.card.isUnit() && event.card.controller !== context.source.controller && context.source.controller.resources.some((resource) => resource.exhausted),
+            },
+            immediateEffect: AbilityHelper.immediateEffects.readyResources({ amount: 1 })
+        });
+    }
+
+    protected override setupLeaderUnitSideAbilities() {
+        this.addTriggeredAbility({
+            title: 'Ready 2 resources',
+            when: {
+                onAttackCompleted: (event, context) => event.attack.attacker === context.source,
+            },
+            immediateEffect: AbilityHelper.immediateEffects.readyResources({ amount: 2 })
```

Comment:
This is missing the check that an enemy left play this phase. This will also be kind of difficult to set up, let me know if you want to tackle it or for me to help.

Take a look at Iden Versio's implementation, specifically the state watcher piece. This is how we record game history. There is currently a state watcher for _defeat_ (the one Iden uses), you would need to add a parallel one for leaves play.

It's not hard if you follow the template, but it's a bit of work just fyi.